sentry-apollo-3/src/main/java/io/sentry/apollo3/SentryApollo3HttpInterceptor.kt

```diff
@@ -1,86 +1,136 @@
 package io.sentry.apollo3
 
+import com.apollographql.apollo3.api.http.DefaultHttpRequestComposer.Companion.HEADER_APOLLO_OPERATION_ID
+import com.apollographql.apollo3.api.http.DefaultHttpRequestComposer.Companion.HEADER_APOLLO_OPERATION_NAME
 import com.apollographql.apollo3.api.http.HttpHeader
 import com.apollographql.apollo3.api.http.HttpRequest
 import com.apollographql.apollo3.api.http.HttpResponse
 import com.apollographql.apollo3.exception.ApolloHttpException
 import com.apollographql.apollo3.exception.ApolloNetworkException
 import com.apollographql.apollo3.network.http.HttpInterceptor
 import com.apollographql.apollo3.network.http.HttpInterceptorChain
-import io.sentry.BaggageHeader
+import io.sentry.BaggageHeader.BAGGAGE_HEADER
 import io.sentry.Breadcrumb
 import io.sentry.Hint
 import io.sentry.HubAdapter
 import io.sentry.IHub
 import io.sentry.ISpan
 import io.sentry.IntegrationName
+import io.sentry.SentryEvent
 import io.sentry.SentryIntegrationPackageStorage
 import io.sentry.SentryLevel
 import io.sentry.SpanStatus
-import io.sentry.TypeCheckHint
+import io.sentry.TypeCheckHint.APOLLO_REQUEST
+import io.sentry.TypeCheckHint.APOLLO_RESPONSE
+import io.sentry.exception.ExceptionMechanismException
+import io.sentry.protocol.Mechanism
+import io.sentry.protocol.Request
+import io.sentry.protocol.Response
+import io.sentry.util.HttpUtils
 import io.sentry.util.PropagationTargetsUtils
 import io.sentry.util.UrlUtils
 import io.sentry.vendor.Base64
+import okio.Buffer
+import org.jetbrains.annotations.ApiStatus
 
-class SentryApollo3HttpInterceptor @JvmOverloads constructor(private val hub: IHub = HubAdapter.getInstance(), private val beforeSpan: BeforeSpanCallback? = null) :
+// TODO: document about not reporting duplicates with ApolloHttpException
+
+class SentryApollo3HttpInterceptor @JvmOverloads constructor(
+    @ApiStatus.Internal private val hub: IHub = HubAdapter.getInstance(),
+    private val beforeSpan: BeforeSpanCallback? = null,
+    private val captureFailedRequests: Boolean = false,
+    private val failedRequestTargets: List<String> = listOf(".*")
+) :
     HttpInterceptor, IntegrationName {
```

Comment:
(l) personal preference, but maybe have the interfaces right next to the colon instead of in a new line?

Reply:
`make format` fixed it.